### DISCUSSION TITLE
Simplification or mistake

### DISCUSSION THREAD

**Discussion by YXY101676** [idaholab/moose#29551] [Q&A Modules: Phase field]
Hello,
The following code defines a function that generates a vector containing the values of order parameters. According to the original formula, this function should sum the squares all order parameters except for the one corresponding to the current kernel. However, based on my understanding, the function is summing the squares of all order parameters:
```
std::vector<Real>
ACGrGrPoly::assignOtherOps()
{
  std::vector<Real> other_ops(_op_num);
  for (unsigned int i = 0; i < _op_num; ++i)
    other_ops[i] = (*_vals[i])[_qp];

  return other_ops;
}

 other_ops = assignOtherOps();

  // Sum all other order parameters
  Real SumOPj = 0.0;
  for (unsigned int i = 0; i < _op_num; ++i)
    SumOPj += other_ops[i] * other_ops[i];
```
I want to know if this usage is a simplification or an error.
yxy

**Comment by GiudGiud** [comment]
Hello

The one for the kernel is not parts of _vals. So it does not get stored in other_ops
see the base class
```
ACGrGrBase::validParams()
{
  InputParameters params = ACBulk<Real>::validParams();
  params.addRequiredCoupledVar("v",
                               "Array of coupled order parameter names for other order parameters");


ACGrGrBase::ACGrGrBase(const InputParameters & parameters)
  : ACBulk<Real>(parameters),
    _op_num(coupledComponents("v")),
    _vals(coupledValues("v")),
```
Then in the SumOPj (really sumOPj_squared), it also does not feature
